src/Plugin/facets/query_type/LocalGovDirectoriesQueryType.php

```diff
@@ -87,4 +125,78 @@ public function build() {
     return $this->facet;
   }
 
+  /**
+   * Get facets from a facet group except it's own with a corresponding tag
+   * 
+   * Re-run the search api query, this time removing all the other facet groups
+   * except the one specified by the filter tag. The purpose of this is to get
+   * the other facets that would be limited by setting of the passed in groups
+   * facets. It allows us to see the other possible facets that could be 
+   * selected as part of an 'OR' group 'AND' the facets selected in the passed
+   * in group.
+   *
+   * @param String $filter_tag
+   *   The search api tag of the facet group.
+   * @return Array
+   *   Search api query facet results.
+   */
+  protected function getResultingFacetsFromFacetGroupExceptOwn(String $filter_tag):Array {
+
+    // Set up a special filter query which needs to clone the original.
+    $filter_query = clone $this->query->getOriginalQuery();
+    $filter_query->preExecute();
+
+    // Find conditions
+    $conditions = &$filter_query->getConditionGroup()->getConditions();
+
+    // Store removed conditions so we can reset them.
+    $removed_conditions = [];
+
+    // Loop through each conditions, removing ones that are not this filter tag.
+    foreach ($conditions as $tag => $condition) {
+      if ($condition instanceof ConditionGroupInterface) {
+        $tags = $condition->getTags();
+
+        // @todo Check that we are only removing facet conditions.
+        if (!in_array($filter_tag, $tags)) {
+
+          // Store the removed conditions and remove it.
+          $removed_conditions[$tag] = $conditions[$tag];
+          unset($conditions[$tag]);
+        }
+      }
+    }
+
+    // Execute the filter query and get the facets returned.
+    $filter_query->execute();
+    $facets = $filter_query->getResults()->getExtraData('search_api_facets');
+    $filter_query->postExecute();
+
+    // Beacuse for reasons unknown, removing the conditions removes it from all
+    // the following queries, even though we are fetching the original query.
```

Comment:
This is because clone only clones the entity itself, references to other entities themselves as references. So it's still a reference to the Condition entity. Thus directly changing the referenced (shared) entity.

One option could be to use DeepCopy that's included in Drupal because (I think of simpletest) but as we don't actually need a fully deep clone can we rather just also clone the Condition and swap that clone on the cloned query?

Reply:
@ekes Would you be able to provide a hint on how to clone just the conditions group for the filter query?
Wouldn't we still need to re-add them though if everything is a clone? I'm not sure how we can actully avoid doing the current method in a way that is not overrly complex.